### PR TITLE
Upgrade PHPSpec and other stuffs

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Upgrading `PHPSpec` version to fix this [issue](https://github.com/phpspec/phpspec/issues/1274).
- Defining different `PHPSpec` versions to support different PHP versions.
The similar approach is on this [PR](https://github.com/php-http/httplug/pull/157/files#).
- Adding `php-7.3` and `php-7.4` version tests on Travis CI build.
- Upgrading cached dependencies with `php-7.0` version because `PHPSpec` version is upgraded.
- Using the `composer update` on Travis CI build for `PHPSpec` version upgrading.
- The `psr-0` autoloading is deprecated. Using the `psr-4` autoloading instead.

### CODE REVIEW

**Comment by peter279k** [comment]
I also think we can consider removing `composer.lock` from Git version control because of different `PHPSpec` versions.

@umpirsky, what do you think about that?

**Comment by umpirsky** [comment]
Yes, let's remove lock file, thanks!

File: .travis.yml [hunk 1]
```diff
@@ -4,9 +4,11 @@ php:
     - 7.0
     - 7.1
     - 7.2
+    - 7.3
+    - 7.4
 
 install:
-    - composer install
+    - composer update
```
Comment (umpirsky) [inline .travis.yml]: Why?

Comment (peter279k) [reply .travis.yml]: Once the `composer.lock` is removed from Git version control, this can be reverted.

Comment (umpirsky) [reply .travis.yml]: Let's do it in this PR?

Comment (peter279k) [reply .travis.yml]: @umpirsky, I've done for this. You can look at latest changes :).

**Comment by umpirsky** [comment]
Thanks!

**Pull request merged**